### PR TITLE
LazyMobXCache promise

### PR DESCRIPTION
**Pull request by adamabeshouse**
Signed-off-by: Abeshouse, Adam A./Sloan Kettering Institute <abeshoua@mskcc.org>

# What? Why?
Fix # .

Changes proposed in this pull request:
- a
- b

# Checks
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). If you are part of the
cBioPortal organization, notify the approprate team (remove inappropriate):

@cBioPortal/frontend

If you are not part of the cBioPortal organization look at who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them here:

### CODE REVIEW

File: src/pages/resultsView/ResultsViewPageStore.ts [hunk 1]
```diff
@@ -633,16 +636,31 @@ export class ResultsViewPageStore {
...
-            }, {});
-        } else {
-            return {};
+    readonly geneToMutationData = remoteData<{[hugoGeneSymbol:string]:CacheData<Mutation[]>|null}>({
+        await:()=>[
+            this.genes,
+            this.mutationDataCache
+        ],
+        invoke:()=>{
+            return this.mutationDataCache.result!.awaitComplete(this.genes.result!, true).then(()=>{
```
Comment (adamabeshouse) [inline src/pages/resultsView/ResultsViewPageStore.ts]: heres where the action is @alisman

Comment (adamabeshouse) [reply src/pages/resultsView/ResultsViewPageStore.ts]: so the second argument says whether you want to passively watch, or watch and also ask for the data to be loaded

Comment (alisman) [reply src/pages/resultsView/ResultsViewPageStore.ts]: What's use case for passively watching?

**Comment by alisman** [comment]
I think this is looking good.   So, how would a view employ this to show the loading status of say, just one gene?

File: src/pages/resultsView/ResultsViewPageStore.ts (hunk 1 above)
Comment (adamabeshouse) [reply src/pages/resultsView/ResultsViewPageStore.ts]: not sure of one at the moment but it doesnt seem obviously useless to me

**Comment by adamabeshouse** [comment]
@alisman e.g. you could set up a remoteData whose invoke simply returns awaitComplete(myOneGene)

File: src/pages/resultsView/ResultsViewPageStore.ts (hunk 1 above)
Comment (adamabeshouse) [reply src/pages/resultsView/ResultsViewPageStore.ts]: actually I think you're right

**Comment by adamabeshouse** [comment]
merged into upcoming oncoprint branch

**Pull request closed without merging**